pkg/common/log/reopen.go

```diff
@@ -0,0 +1,110 @@
+package log
+
+import (
+	"context"
+	"io"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+)
+
+const (
+	_fileFlags    = os.O_APPEND | os.O_CREATE | os.O_WRONLY
+	_fileMode     = 0640
+	_reopenSignal = syscall.SIGUSR2
```

Comment:
Also, `syscall.SIGUSR2` (and in general, this functionality) isn't available on windows. We've just undergone an effort to get us building under windows, so we'll need to make this functionality a no-op on windows.

Reply:
Ack on the style nit - will fix - I'm coming from a different school.

OK. We can noop for windows. Maybe it should spit out a warning in the log  that it cannot rotate when `log_file` is set under windows? Also, who runs `logrotate` on windows?

Reply:
I don't think folks running on windows will have an expectation of signal based log rotation, so I'd be ok leaving out the warning.

Reply:
Ack.

Reply:
Per further discussion, implementing no op for windows will involve build tags and alternate code because `syscall.SIGUSR2` is not defined for windows. I will need to find or make an environment where I can confirm it builds under windows.

Reply:
@zmt, I'm happy to help with the Windows stuff. May I add a commit on top of this branch with that?

Reply:
yes - thanks - merged your PR to my PR just now